### DISCUSSION TITLE
Gauge with data per environment

### DISCUSSION THREAD

**Discussion by sampadrout** [getgauge/gauge#2029] [General]
Hi, i want to run same set of scenarios agains 3 different environments thats dev, qa and pre-prod. Can you please provide me some example how i can achieve this using javascript. Thanks in advance.

**Comment by zabil** [comment]
Please refer https://docs.gauge.org/configuration.html?os=macos&language=javascript&ide=vscode#using-environments-in-a-gauge-project

**Reply by sampadrout** [reply]
@zabil thanks for the link. I am interested to know how to pick a parameter "csv_file" from the command line "gauge run --env qa specs" and then fetch the value of that parameter from the properties file like csv_file=resources/qa/users.csv and provide it to the step as: 
* Log into wordpress app as user <table:csv_file>

**Reply by zabil** [reply]
That's not possible at the moment. There is a discussion about this at https://github.com/getgauge/gauge/issues/1518

**Comment by sampadrout** [comment]
ok. Let me go ahead with this alternate approach. First capture the csv_file= resources/qa/csv_file.csv as per the environment and then copy the files using a hook to a predefined path to which table: resources/csv_file.csv pointing to:
const srcDir = process.env.csv_file;
const destDir = 'resources/';

 fs.copySync(srcDir, destDir, function (err) {
            if (err) return console.error(err)
            console.log('Files copy success!')
        });
